sos/plugins/watchdog.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (C) 2018 Red Hat, Inc., Reid Wahl <nwahl@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+'''Collect info from watchdog to add to sosreport.
+
+Classes:
+Watchdog -- collects watchdog configs, custom executables, and logs
+'''
+
+import re
+
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class Watchdog(Plugin, RedHatPlugin):
+    '''Collect watchdog configurations, custom executables, and logs.
+
+    Methods:
+    parse_watchdog_conf -- parse variables from config file
+    setup -- assemble watchdog info for sosreport
+
+    Instance variables:
+    limit -- maximum log size
+    watchdog_conf -- watchdog service configuration file
+    conf_vars -- variables from watchdog config file
+    log_dir -- directory containing watchdog stderr and stdout logs
+    '''
+
+    plugin_name = 'watchdog'
+    profiles = ('system',)
+    packages = ('watchdog',)
+
+    def __init__(self, commons):
+        '''Initialize Watchdog class.'''
+        Plugin.__init__(self, commons)
+        self.limit = self.get_option('log_size')
+        self.watchdog_conf = '/etc/watchdog.conf'
+        self.conf_vars = dict()
+        self.log_dir = None
+
+    def parse_watchdog_conf(self):
+        '''Parse variables from watchdog config file.'''
+        comment_pattern = re.compile(r'#.*')
```

Comment:
Somewhat crude way to remove comments at end of line rather than simply skipping lines that begin with a comment. This does not guard against removing a literal `#` character in a quoted string. However, to my knowledge there is no reason a literal `#` would be used in a value in `/etc/watchdog.conf`. So I chose not to make the comment parsing any more complicated than this.